pkg/apis/resource/types.go

```diff
@@ -323,20 +323,6 @@ type ResourceClaimSpec struct {
 	//
 	// +optional
 	Devices DeviceClaim
-
-	// Controller is the name of the DRA driver that is meant
-	// to handle allocation of this claim. If empty, allocation is handled
-	// by the scheduler while scheduling a pod.
-	//
-	// Must be a DNS subdomain and should end with a DNS domain owned by the
-	// vendor of the driver.
-	//
-	// This is an alpha field and requires enabling the DRAControlPlaneController
-	// feature gate.
-	//
-	// +optional
-	// +featureGate=DRAControlPlaneController
-	Controller string
```

Comment:
A question, so I am clear how this work. Once we upgrade, existing etcd record stored in the v1alpha3 format can be read, but they won't be written back in that format, and any data in these fields that have been removed will be lost. This is OK, because we do not guarantee round tripping of alphas forever (like, in this case when they are removed). Is that accurate?

Reply:
> Once we upgrade, existing etcd record stored in the v1alpha3 format can be read, but they won't be written back in that format

Currently, they will be written as v1alpha3, but without these fields.

In my beta1 branch, I bump the storage version to v1beta1.

> This is OK, because we do not guarantee round tripping of alphas forever (like, in this case when they are removed). Is that accurate?

Yes. Downgrades from 1.32 to 1.31 will not be supported.